extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/QuarkusResourceProvider.java

```diff
@@ -0,0 +1,56 @@
+package io.quarkus.flyway.runtime;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.flywaydb.core.api.ResourceProvider;
+import org.flywaydb.core.internal.resource.LoadableResource;
+import org.flywaydb.core.internal.util.StringUtils;
+import org.jboss.logging.Logger;
+
+/**
+ * This implementations of this class are very similar to those of {@link org.flywaydb.core.internal.scanner.Scanner}
+ * TODO: refactor upstream to move methods to utility class
+ */
+public class QuarkusResourceProvider implements ResourceProvider {
```

Comment:
I wonder if I would call the classes `QuarkusFlyway(Resource|Class)Provider` to be extra clear.

Reply:
Sure thing